ext/reflection/php_reflection.c

```diff
@@ -6116,33 +6116,32 @@ static zend_bool is_ignorable_reference(HashTable *ht, zval *ref) {
 	return Z_TYPE_P(Z_REFVAL_P(ref)) != IS_ARRAY || Z_ARRVAL_P(Z_REFVAL_P(ref)) != ht;
 }
 
-/* {{{ proto public ReflectionReference|null ReflectionReference::fromArrayElement(array array, mixed key)
+/* {{{ proto public ReflectionReference|null ReflectionReference::fromArrayElement(array array, string|int key)
  *     Create ReflectionReference for array item. Returns null if not a reference. */
 ZEND_METHOD(ReflectionReference, fromArrayElement)
 {
-	HashTable *ht;
-	zval *key, *item;
+	zval *item, *array;
+	zend_string *string_key = NULL;
+	zend_long int_key = 0;
 	reflection_object *intern;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "hz", &ht, &key) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_ARRAY(array)
```

Comment:
```suggestion
		Z_PARAM_ARRAY_HT(ht)
```
to avoid unnecessary changes.